components/monitoring/README.md

```diff
@@ -0,0 +1,141 @@
+        name: prometheus
+        labels:
+          prometheus: appstudio-workload
+      spec:
+        endpoints:
+        - bearerTokenSecret:
+            key: ""
+          interval: 15s
+          path: /metrics
+          port: oauth2-proxy
+          scheme: HTTPS
+          tlsConfig:
+            caFile: /etc/prometheus/tls/tls.crt 
+            serverName: "prometheus-oauth2.appstudio-workload-monitoring.svc"
+        namespaceSelector:
+          matchNames:
+          - appstudio-workload-monitoring
+        selector:
+          matchLabels:
+            app.kubernetes.io/instance: monitoring-workload-in-cluster
+      ```
+     
+      
+
+  - If the servicemonitor is for getting other components added to prometheus monitoring
+      
+      - Add a sevicemonitor decalration for scrapping the intended service
+      
+          ```yaml
+          apiVersion: monitoring.coreos.com/v1
+          kind: ServiceMonitor
+          metadata:
+            labels:
+              prometheus: appstudio-workload <label getting discovered by prometheus-operator>
+            name: release-service
+            namespace: release-service `<name of the namespace the service is in>`
+          spec:
+            endpoints:
+            - path: /metrics
+              port: https
+              scheme: https
+              bearerTokenSecret:
+                name: `<secret for accessing the endpoint path on the service>`
+                key: token
+              tlsConfig:
+                insecureSkipVerify: true
+            namespaceSelector:
+              matchNames:
+              - release-service
+            selector:
+              matchLabels:
+                control-plane: controller-manager
+          ```
+
+      - Note: `make sure the namespace` is the namespace for the intended service itself, in this case `release-service`.
+
+    b. It should have the accessible port and route to the service (or service url)
+
+    c. Access token or service accounts as required.
+
+    d. Use label selectors to select the desired service uniquely in the cluster.
+
+### 3. View access to the exporter service for Prometheus
+
+    a. Prometheus should have view access to the metrics exporter service namespace
+
+    b. Add the Rolebinding to give prometheus view access. 
+  - [Here](https://github.com/redhat-appstudio/infra-deployments/blob/main/components/monitoring/prometheus/base/prometheus-view.yaml) is an example providing prometheus view access to the cluster
+
+  ```yaml
+  kind: ClusterRoleBinding
+  apiVersion: rbac.authorization.k8s.io/v1
+  metadata:
+    name: prometheus-view
+  subjects:
+  - kind: ServiceAccount
+    name: prometheus-k8s
+    namespace: appstudio-workload-monitoring
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: view
+  ```
+
+#### 4. Grafana dashboards - manual export
+
+    a. Create a new folder in grafana for your service. (In the left nav, + Create Folder)
```

Comment:
Which grafana? Where?

Reply:
added link there for the stage deployment.